.github/workflows/check.yaml

```diff
@@ -178,6 +178,49 @@ jobs:
       - name: Build example app
         run: swift run BuildTool build-example-app --platform ${{ matrix.platform }}
 
+  check-documentation:
+    runs-on: macos-15
+
+    permissions:
+      deployments: write
+      id-token: write
+      contents: read
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      # This step can be removed once the runners’ default version of Xcode is 16 or above
+      - uses: maxim-lobanov/setup-xcode@v1
+        with:
+          xcode-version: 16
+
+      # Build the documentation using Swift DocC
+      - name: Build documentation
+        run: |
+          PR_NUMBER=${{ github.event.pull_request.number }}
+          swift package generate-documentation --target AblyChat --disable-indexing \
+          --hosting-base-path "/builds/ably/ably-chat-swift/pull/${PR_NUMBER}/AblyChat"
```

Comment:
unfortunately it won't work for main, there is some logic hidden in `ably/sdk-upload-action`, it generates different path for based on trigger, for main branche it will be `/main`, for prs it `/pull/${PR_NUMBER}`, and something different for tags.

it's uglier of course but we can upload empty dir first and get url from `ably/sdk-upload-action`:

```yaml
     - name: Dry-Run Upload (to get url) 
        id: preupload
        uses: ably/sdk-upload-action@v2
        with:
          sourcePath: SOME_EMPTY_DIR (we can generate it on previous step)
          githubToken: ${{ secrets.GITHUB_TOKEN }}
          artifactName: AblyChat
```

or we can try to use `--hosting-base-path "./"` maybe it will work

Reply:
added dry run.. marking as resolved